src/scipp/io/hdf5.py

```diff
@@ -396,7 +396,10 @@ def read(group):
 
 
 def _direct_io(cls, convert=None):
-    type_name = cls.__name__
+    if cls is np.bool:
+        type_name = 'npbool'
+    else:
+        type_name = cls.__name__
```

Comment:
I don't like this hack, with numpy 2.0 booleans are back to `np.bool` https://numpy.org/devdocs/release/2.0.0-notes.html#changes and hence the name dunder will return `bool`, the same as inbuilt python `bool`. The hdf5 writer would treat both `np.bool` and `bool` same. We can look at explicit type instead of the `name` dunder but it will be a bit more invasive than this change.

Reply:
Let's avoid hacks! If using classes directly instead of names is cleaner, do that!

Reply:
We need to keep some parts of this hacky behaviour to support both 1.x and 2.x, I have added a TODO instead.